File: block_hosts/biubiu/src/main/java/com/leichao/biubiu/SplashActivity.java
package com.leichao.biubiu;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import com.leichao.util.ToastUtil;
import com.qihoo360.replugin.RePlugin;

public class SplashActivity extends AppCompatActivity {

    TextView tvSplash;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        tvSplash = findViewById(R.id.splash_text);

        tvSplash.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ToastUtil.show("启动页");
            }
        });

        new Thread(new Runnable() {
            @Override
            public void run() {
                final String pluginName = "biubiu_home";
                if (RePlugin.preload(pluginName)) {
                    tvSplash.post(new Runnable() {
                        @Override
                        public void run() {
                            RePlugin.startActivity(SplashActivity.this,
                                    RePlugin.createIntent(pluginName, "com.leichao.biubiu.home.HomeActivity"));
                            finish();
                        }
                    });
                }
            }
        }).start();
    }

}

File: block_hosts/biubiu/src/main/java/com/leichao/biubiu/SplashActivity.kt
package com.leichao.biubiu

import com.leichao.common.base.BaseActivity
import com.leichao.common.plugin.PluginConstant
import com.leichao.util.StatusBarUtil
import com.leichao.util.ToastUtil
import com.qihoo360.replugin.RePlugin
import kotlinx.android.synthetic.main.activity_splash.*

class SplashActivity : BaseActivity() {

    override fun initView() {
        setContentView(R.layout.activity_splash)
        StatusBarUtil.setFullTranslucent(this)
        StatusBarUtil.setTextColor(this, true)
    }

    override fun initData() {
        val startTime = System.currentTimeMillis()
        Thread(Runnable {
            val pluginName = PluginConstant.BIUBIU_HOME
            if (RePlugin.preload(pluginName)) {
                val delayMillis = if (System.currentTimeMillis() - startTime > 1000L) 0L else 1000L
                splash_text.postDelayed({
                    val activities = RePlugin.fetchPackageInfo(pluginName).activities
                    if (activities != null && activities.isNotEmpty()) {
                        RePlugin.startActivity(this@SplashActivity,
                                RePlugin.createIntent(pluginName, activities[0].name))
                    }
                    finish()
                }, delayMillis)
            }
        }).start()
    }

    override fun initEvent() {
        splash_text.setOnClickListener { ToastUtil.show("启动页") }
    }

}
